ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/BlockPropagationManager.java

```diff
@@ -292,29 +276,61 @@ private void handleNewBlockHashesFromNetwork(final EthMessage message) {
       return CompletableFuture.completedFuture(block);
     }
 
-    ethContext.getScheduler().scheduleSyncWorkerTask(() -> validateAndBroadcastBlock(block));
+    final BlockHeader parent =
+        protocolContext
+            .getBlockchain()
+            .getBlockHeader(block.getHeader().getParentHash())
+            .orElseThrow(
+                () ->
+                    new IllegalArgumentException(
+                        "Incapable of retrieving header from non-existent parent of "
+                            + block.getHeader().getNumber()
+                            + "."));
+
+    final ProtocolSpec<C> protocolSpec =
+        protocolSchedule.getByBlockNumber(block.getHeader().getNumber());
+    final BlockHeaderValidator<C> blockHeaderValidator = protocolSpec.getBlockHeaderValidator();
 
     // Import block
     final PersistBlockTask<C> importTask =
         PersistBlockTask.create(
-            protocolSchedule, protocolContext, block, HeaderValidationMode.FULL, metricsSystem);
+            protocolSchedule, protocolContext, block, HeaderValidationMode.NONE, metricsSystem);
+
     return ethContext
         .getScheduler()
-        .scheduleSyncWorkerTask(importTask::run)
-        .whenComplete(
-            (r, t) -> {
-              importingBlocks.remove(block.getHash());
-              if (t != null) {
+        .scheduleSyncWorkerTask(
+            () -> {
+              if (blockHeaderValidator.validateHeader(
+                  block.getHeader(), parent, protocolContext, HeaderValidationMode.FULL)) {
+                ethContext
+                    .getScheduler()
+                    .scheduleSyncWorkerTask(() -> broadcastBlock(block, parent));
+                return ethContext
```

Comment:
I'd be tempted to extract this blob that does the actual import into a separate method as well.